src/IO/tcp/SocketServer.cpp

```diff
@@ -79,10 +79,14 @@ void SocketServer::sendingLoop()
             auto data = sendingBuffer.front();
 
             std::lock_guard<std::mutex> lockGuard(clientsMutex);
-            for (auto &client : clients)
-            {
-                client->send(data);
+            
+            if (clients.size() > 0 && clients[clients.size() - 1] != nullptr) {
+                clients[clients.size() - 1]->send(data);
             }
+            //for (auto &client : clients)
+            //{
+            //    client->send(data);
+            //}
```

Comment:
We can undo this change, but other than that, I think it is good to merge

Reply:
Yeah, we need to debug and rework that part of the code.